ui/app/components/calendar-widget.js

```diff
@@ -0,0 +1,176 @@
+  @tracked disableFutureYear = this.isCurrentYear(); // if current year, disable right chevron
+  @tracked endDateDisplay;
+  @tracked showCalendar = false;
+  @tracked showSingleMonth = false;
+
+  constructor() {
+    super(...arguments);
+    // ARG TODO we need to return the config's duration if not default to 12 months to calculate, now it's 12 months
+    let date = new Date();
+    date.setMonth(date.getMonth() - 1); // by default you calculate the end month as the month prior to the current month.
+    this.endDateDisplay = format(date, 'MMMM yyyy');
+  }
+
+  // HELPER FUNCTIONS (alphabetically) //
+  addClass(element, classString) {
+    element.classList.add(classString);
+  }
+
+  isCurrentYear() {
+    return this.currentYear === this.displayYear;
+  }
+
+  isObsoleteYear() {
+    // do not allow them to choose a end year before the this.args.startDate
+    let startYear = this.args.startDate.split(' ')[1];
+    return this.displayYear.toString() === startYear; // if on startYear then don't let them click back to the year prior
+  }
+
+  removeClass(element, classString) {
+    element.classList.remove(classString);
+  }
+
+  // ACTIONS (alphabetically) //
+  @action
+  addYear() {
+    this.displayYear = this.displayYear + 1;
+    this.disableMonths();
+    this.disableFutureYear = this.isCurrentYear();
+    this.disablePastYear = this.isObsoleteYear();
+  }
+
+  @action
+  disableMonths() {
+    this.allMonthsNodeList = document.querySelectorAll('.is-month-list');
+    this.allMonthsNodeList.forEach((e) => {
+      // clear all is-readOnly classes and start over.
+      this.removeClass(e, 'is-readOnly');
+
+      let elementMonthId = parseInt(e.id.split('-')[0]); // dependent on the shape of the element id
+      // for current year
+      if (this.currentMonth <= elementMonthId) {
+        // only disable months when current year is selected
+        if (this.isCurrentYear()) {
+          e.classList.add('is-readOnly');
+        }
+      }
+      // compare for startYear view
+      if (this.displayYear.toString() === this.args.startDate.split(' ')[1]) {
+        // if they are on the view where the start year equals the display year, check which months should not show.
+        let startMonth = this.args.startDate.split(' ')[0]; // returns month name e.g. January
+        // return the index of the startMonth
+        let startMonthIndex = MONTH_MAPPING.indexOf(startMonth);
+        // then add readOnly class to any month less than the startMonth index.
+        if (startMonthIndex > elementMonthId) {
+          e.classList.add('is-readOnly');
+        }
+      }
+    });
+  }
+
+  @action
+  selectCurrentBillingPeriod() {
+    // ARG TOOD send to dashboard the select current billing period. The parent may know this it's just a boolean.
+    // Turn the calendars off if they are showing.
+    this.showCalendar = false;
+    this.showSingleMonth = false;
+  }
+  @action
+  selectEndMonth(month, year, e) {
```

Comment:
what is `e` supposed to be? A more descriptive param name would be helpful :)

Reply:
good call. it's the element, so the specific month selected. In this particular case we need to filter through the months using the index and return that element when they match.

modified e to element to clarify.